pkg/chunkenc/symbols.go

```diff
@@ -102,15 +105,15 @@ func (s *symbolizer) add(lbl string) uint32 {
 }
 
 // Lookup coverts and returns labels pairs for the given symbols
-func (s *symbolizer) Lookup(syms symbols, buf labels.Labels) labels.Labels {
+func (s *symbolizer) Lookup(syms symbols, buf *log.BufferedLabelsBuilder) labels.Labels {
 	if len(syms) == 0 {
-		return nil
+		return labels.EmptyLabels()
 	}
+
 	if buf == nil {
-		buf = structuredMetadataPool.Get().(labels.Labels)
+		structuredMetadata := structuredMetadataPool.Get().(labels.Labels)
+		buf = log.NewBufferedLabelsBuilder(structuredMetadata)
 	}
-	buf = buf[:0]
```

Comment:
Do we not need to clear the BufferedLabelBuilder?

Reply:
That's what `BufferedLabelsBuilder` does on creation.